libraries/redcore/api/hal/hal.php

```diff
@@ -1220,6 +1244,11 @@ public function processPostData($data, $configuration)
 				$fieldAttributes['defaultValue'] = !empty($fieldAttributes['defaultValue']) ? $fieldAttributes['defaultValue'] : '';
 				$data[$fieldAttributes['name']] = !empty($data[$fieldAttributes['name']]) ? $data[$fieldAttributes['name']] : $fieldAttributes['defaultValue'];
 				$data[$fieldAttributes['name']] = $this->transformField($fieldAttributes['transform'], $data[$fieldAttributes['name']], false);
+
+				if ($removeNulls && $data[$fieldAttributes['name']] == '')
```

Comment:
I thought of that too, we'd need to have one per attribute ofc.

However, the unset is not because of the transform, but because of mysql.  I discovered while testing that the problem are not the transform functions, the problem is that mysql is converting ''  to 0 for ints, so there's actually no way to set null unless we do that isNull per field.

However, isn't that an overkill?